extensions/canvasMedical/validation/dto/appointment.zod.ts

```diff
@@ -0,0 +1,86 @@
+import { z } from 'zod'
+import { createReferenceSchema } from './reference.zod'
+
+const reasonCodeSchema = z.object({
+  coding: z.array(
+    z.object({
+      system: z.literal('INTERNAL'),
+      code: z.string(),
+      display: z.string(),
+    })
+  ),
+  text: z.string(),
+})
+
+const participantSchema = z.object({
+  actor: z.object({
+    reference: z.union([
+      createReferenceSchema('Practitioner'),
+      createReferenceSchema('Patient'),
+    ]),
+  }),
+  status: z.literal('accepted'),
+})
+
+const containedSchema = z.object({
+  resourceType: z.literal('Endpoint'),
+  id: z.string(),
+  status: z.string(),
+  connectionType: z.object({
+    code: z.string(),
+  }),
+  payloadType: z.array(
+    z.object({
+      coding: z.array(
+        z.object({
+          code: z.string(),
+        })
+      ),
+    })
+  ),
+  address: z.string(),
+})
+
+export const appointmentSchema = z.object({
+  resourceType: z.literal('Appointment'),
+  reasonCode: z.array(reasonCodeSchema).optional(),
+  description: z.string().optional(),
+  participant: z.array(participantSchema).optional(),
+  appointmentType: z.object({
+    coding: z.array(
+      z.object({
+        system: z.literal('http://snomed.info/sct'),
+        code: z.enum([
+          '439708006',
+          '448337001',
+          '308335008',
+          '31108002',
+          '185317003',
+        ]),
```

Comment:
@kubiak85 where is this from?

Reply:
@rahulkeerthi I checked the docs and indeed there are some values hardcoded https://docs.canvasmedical.com/reference/appointment-create#appointmenttype, but it is probable that we shouldn't validate it and let Canvas return error when there is no match, because those values may possibly change